portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -25,24 +25,25 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  private List<String> comments;
+  private List<String> comments = new ArrayList<>();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    comments = new ArrayList<>();
-    comments.add("These");
-    comments.add("are");
-    comments.add("test");
-    comments.add("comments");
-
     String json = "{\"comments\": [";
     for (String comment : comments)
-      json = json.concat("{\"comment\": \"" +  comment + "\" },");
+      json = json.concat("{\"value\": \"" +  comment + "\" },");
     //replaces last comma with JSON enclosing bracket
     json = "".concat(json.substring(0, json.length() - 1) + "]}");
-
     
     response.setContentType("application/json;");
     response.getWriter().println(json);
+    response.sendRedirect("/index.html");
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String comment = request.getParameter("Comment:");
```

Comment:
Hmm, this will depend on how you form your request on the client side but you probably don't need the colon in this string.